File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        vector<int> ha(26, 0);
        vector<int> hb(26, 0);
        for(auto c : s1)
            ha[c-'a']++;
        int k = s1.length();
        int i=0, n=s2.length();
        for(int j=0; j<n; j++)
        {
            hb[s2[j]-'a']++;

            if(j-i > k-1)
                --hb[s2[i++]-'a'];
            if(ha == hb)
                return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-0.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut finder = HashMap::new();

        for (index, &num) in nums.iter().enumerate() {
            let y = target - num;
            if let Some(&j) = finder.get(&y) {
                return vec![j as i32, index as i32];
            }
            finder.insert(num, index);
        }
        vec![]
    }
}